base/shell/explorer/explorer.cpp

```diff
@@ -87,6 +87,43 @@ HideMinimizedWindows(IN BOOL bHide)
 }
 #endif
 
+static BOOL
+IsExplorerSystemShell()
+{
+    HKEY hKeyWinlogon;
+	WCHAR szShell[256];
+	WCHAR szPath[MAX_PATH];
+    LPWSTR szExplorer = NULL;
+	DWORD dwType;
+	DWORD dwBufferSize = sizeof(szShell);
+
+    if (!GetModuleFileName(NULL, szPath, MAX_PATH))
+        return FALSE;
+
+    szExplorer = PathFindFileName((LPWSTR)szPath);
+
+	if (RegOpenKeyEx(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+		0, KEY_READ, &hKeyWinlogon) != ERROR_SUCCESS)
+    {
+        // No registry access.
+        return TRUE;
+    }
+
+    if (RegQueryValueEx(hKeyWinlogon, L"Shell", 0, &dwType,
+		(LPBYTE)szShell, &dwBufferSize) == ERROR_SUCCESS)
+	{
+		RegCloseKey(hKeyWinlogon);
```

Comment:
RegCloseKey needs to happen even if RegQueryValueEx fails. Why not store the return value in a BOOL initialized to TRUE?